Console/retro/Console.h

```diff
@@ -83,17 +86,23 @@ namespace retro
         short GetRows() const { return rows; }
         short GetCols() const { return cols; }
 
+        virtual void setWindowName(std::string newName) {};
+
+
         void Idle();
 
         bool IsEOF() const { return eof; }
+
     private:
+        std::string windowName;
         GrafPtr consolePort = nullptr;
         Rect bounds;
         Attributes currentAttr;
 
         std::vector<AttributedChar> chars, onscreen;
         bool isProcessingEscSequence;
         int sequenceStep;
+        bool OSCseq;
```

Comment:
The "state machine" here is starting to get a bit complex now... the state is represented by 3 separate variables (`isProcessingEscSequence`, `sequenceStep`, and `OSCseq`) that are all changed in different places.
I have a suspicion that things might become cleaner if you write one big `enum` and give each state a name, something like:
```
enum class State
{
    noSequence,
    waitingForSequenceStart, 
    waitingForStyle,
    waitingForM,
    waitingForOSC0,
    inWindowName
};
```

My names are bad, for perfection double-check the Internet for the "official" names for the various parts of an escape sequence. Then everything would be just one switch, you'd have to explicitly mention the "next" state instead of using `++sequenceStep`, but you could also just write `state = State::noSequence;` instead of three lines resetting all three variables.

Alternatively, you could have something like
```
enum class State
{
    noSequence,
    characterStyleSequence,
    oscSequence
}
```
just to make things more symmetric (and get rid of one variable) and still have `sequenceStep` as a separate variable.
I'm leaning towards the first, but I'd need to try it out in order to know which one ends up more beautiful.

Reply:
I like very much state machines, I tried and it seems to me that this solution works extremely well.